matminer/featurizers/site.py

```diff
@@ -804,15 +801,19 @@ def fit(self, structs):
         in a dict of self.el_amt_dict_, avoiding repeated calculation of
         composition when featurizing multiple sites in the same structure.
         Args:
-            structs (pandas series): series of pymatgen Structures.
+            X (pandas series): series of pymatgen Structures.
```

Comment:
this should just be a list-like object of pymatgen structures (not necessarily a pandas series)

Reply:
also is this correct? do you need a 2D array of structures and sites? if the latter the docstring should read:

```
X (2D array): 2D array of pymatgen Structures and Sites. One can also pass two dataframe columns, e.g. df["structures", "sites"]
```

Not sure if what I wrote above is correct, but please double-check your docs

Reply:
Also it's unclear to me what the format of X should be if one is not using dataframes - the tests don't have any examples. Can you clarify the non-dataframe format of X?

Reply:
Forgot to cleanup the docs... I will fix this soon. 

The input X now has multiple choices: 
1. X can be 2D list-like object: `[['structure', 'site'], ['structure', 'site'], …]` for instance
2. X can be dataframe:  `df[['structure', 'site']]`
3. X can be tolerant to 1D object: `[structure, site]`.

They will all be transformed to `np.array` within `np.atleast2D()`. Is this behavior desirable?

The test of `df[['structure', 'site']].values` is embedded in the test of `fit_transform` currently. I will update the tests and also add test of a 1D object later.